internal/metamorph/server.go

```diff
@@ -438,6 +448,17 @@ func (s *Server) GetTransactionStatus(ctx context.Context, req *metamorph_api.Tr
 		RejectReason: data.RejectReason,
 		CompetingTxs: data.CompetingTxs,
 		MerklePath:   data.MerklePath,
+		Timestamp:    timestamppb.New(data.StoredAt),
```

Comment:
It is not the `storedAt` field that we want to compare with. It is the field `LastSubmitted`